internal/api/httpsrv/handler_oauth_client.go

```diff
@@ -2,22 +2,77 @@ package httpsrv
 
 import (
 	"context"
+	"net/http"
 
-	v1 "go.infratographer.com/identity-api/pkg/api/v1"
+	"go.infratographer.com/identity-api/internal/crypto"
+	"go.infratographer.com/identity-api/internal/types"
 )
 
-func (h *apiHandler) DeleteOAuthClient(ctx context.Context, request DeleteOAuthClientRequestObject) (DeleteOAuthClientResponseObject, error) {
-	return DeleteOAuthClient200JSONResponse{Success: true}, nil
+const defaultTokenLength = 26
+
+// CreateOAuthClient creates a client for a tenant with a set name.
+// This endpoint returns the OAuth client ID and secret that the client
+// needs to provide to authenticate when requesting a token.
+func (h *apiHandler) CreateOAuthClient(ctx context.Context, request CreateOAuthClientRequestObject) (CreateOAuthClientResponseObject, error) {
+	var newClient types.OAuthClient
+	newClient.TenantID = request.TenantID.String()
+
+	if request.Body.Audience != nil {
+		newClient.Audience = *request.Body.Audience
+	}
```

Comment:
`Name` isn't being set here. Also, `audience` is `null` on the create response but `[]` on get:

```vscode ➜ ~ $ curl -s -XPOST -H 'Content-Type: application/json' -d '{"name": "cool client"}' http://localhost:8000/api/v1/tenants/67787b34-866e-4b75-a395-5aba096b2c1b/clients | jq
{
  "audience": null,
  "id": "c789d22e-9a06-4629-804c-979cd0f43d3b",
  "name": "",
  "secret": "zZyzO5IuR69iQnKVi3Qhi3iaOr"
}
vscode ➜ ~ $ curl -s http://localhost:8000/api/v1/clients/c789d22e-9a06-4629-804c-979cd0f43d3b | jq
{
  "audience": [],
  "id": "c789d22e-9a06-4629-804c-979cd0f43d3b",
  "name": "",
  "secret": ""
}```